src/frontend/src/optimizer/plan_node/logical_project.rs

```diff
@@ -264,7 +265,12 @@ impl ToStream for LogicalProject {
         let (proj, out_col_change) = self.rewrite_with_input(input.clone(), input_col_change);
 
         // Add missing columns of input_pk into the select list.
-        let input_pk = input.stream_key();
+        let input_pk = input.stream_key().unwrap_or_else(|| {
+            panic!(
+                "should always have a stream key in the stream plan but not, sub plan: {}",
+                input.explain_to_string()
+            )
+        });
```

Comment:
Seems to have too many same `unwrap_or_else`s, may extract to `PlanRef::expect_stream_key`?

Reply:
will do it later